python/TestHarness/TestHarness.py

```diff
@@ -794,6 +794,14 @@ def writeResults(self):
 
                 # Create empty key based on TestDir, or re-inialize with existing data so we can append to it
                 self.options.results_storage[job.getTestDir()] = self.options.results_storage.get(job.getTestDir(), {})
+
+                # If output has been stored in separate files, don't make additional copies by
+                # storing that data in this json results file (--pbs || --sep-files, etc options).
+                if job.getOutputFile() is not None:
+                    output = ''
+                else:
+                    output = job.getOutput()
```

Comment:
Non-actionable, but just for future reference -- anything that evaluates to None in a python conditional is considered "False-y", and anything that returns a value will evaluate as True. So the pythonic way to do this is 

```suggestion
                output = '' if job.getOutputFile() else job.getOutput()
```

The exception to this is if like, getOutputFile is specifically supposed to return True or False, and you are checking if it is None, but if it isn't, the way i described is the most pythonic

You do this in other parts of the PR so you probably already know this, and it doesn't really matter anyway.

Reply:
I like it